cmux.go

```diff
@@ -177,7 +177,8 @@ func (l muxListener) Accept() (net.Conn, error) {
 // MuxConn wraps a net.Conn and provides transparent sniffing of connection data.
 type MuxConn struct {
 	net.Conn
-	buf buffer
+	buf     bytes.Buffer
```

Comment:
I wouldn't call that a simplification, you're just pushing the complexity to bufferedReader. At that point, why even have two structs?